### PR TITLE
TRUNK-5067 Add tests to MessageServiceTest, added test cases for bran…

### PR DESCRIPTION
**Pull request by dilekamadushan**
https://issues.openmrs.org/browse/TRUNK-5067?filter=-1
Description of what I changed

TRUNK-5067 Add tests to MessageServiceTest, added test cases for branches, MessagePreparators in MessageServiceTest.java

Checklist: I completed these to help reviewers :)

[ x] My pull request only contains ONE single commit

[x ] My IDE is configured to follow the code style of this project.

[ x] I have added tests to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

[x ] I ran mvn clean package right before creating this pull request and
added all formatting changes to my commit.

[ x] All new and existing tests passed.

[ x] My pull request is based on the latest changes of the master branch.

### CODE REVIEW

**Comment by dilekamadushan** [comment]
@dkayiwa Feel free to review on my PR

File: api/src/test/java/org/openmrs/notification/MessageServiceTest.java [hunk 1]
```diff
@@ -113,4 +115,25 @@ public void sendMessage_shouldSendMessage() throws MessageException {
 		}
 	}
 	
+	/**
+	 * @see MessageService#getMessagePreparator()
+	 */
+	@Test
+	public void getMessagePreparator_shouldGetMessagePreparator() {
+		
+		assertNotNull( ms.getMessagePreparator() );
```
Comment (f4ww4z) [inline api/src/test/java/org/openmrs/notification/MessageServiceTest.java]: Remove the spaces. E.g. `assertNotNull(ms.getMessagePreparator());`

File: api/src/test/java/org/openmrs/notification/MessageServiceTest.java [hunk 2]
```diff
@@ -113,4 +115,25 @@ public void sendMessage_shouldSendMessage() throws MessageException {
...
+	}
+	
+	/**
+	 * @see MessageService#setMessagePreparator(MessagePreparator)
+	 */
+	@Test
+	public void setMessagePreparator_shouldSetMessagePreparator(){
+		
+		ms.setMessagePreparator( null);
+		assertNull( ms.getMessagePreparator() );
```
Comment (f4ww4z) [inline api/src/test/java/org/openmrs/notification/MessageServiceTest.java]: Remove the spaces. E.g. `ms.setMessagePreparator(null);` and `assertNull(ms.getMessagePreparator());`

**Comment by dilekamadushan** [comment]
@f4ww4z I modified the code on your review. Thanks for your reveiws, hoping for more.

**Comment by dilekamadushan** [comment]
@f4ww4z ok I will work on some more tests. Thanks for support

**Comment by dilekamadushan** [comment]
@f4ww4z I added one more test case. Feel free to review. Thanks

**Comment by dilekamadushan** [comment]
@f4ww4z Thanks for your support. Feel free to review on #2541 and #2549

File: api/src/test/java/org/openmrs/notification/MessageServiceTest.java [hunk 3]
```diff
@@ -113,4 +118,107 @@ public void sendMessage_shouldSendMessage() throws MessageException {
 		}
 	}
 	
+	/**
+	 * @see MessageService#getMessagePreparator()
+	 */
+	@Test
+	public void getMessagePreparator_shouldGetMessagePreparator() {
```
Comment (teleivo) [inline api/src/test/java/org/openmrs/notification/MessageServiceTest.java]: remove this one, there is no logic in the getter. we dont test such getters. same applies for setters.

File: api/src/test/java/org/openmrs/notification/MessageServiceTest.java [hunk 4]
```diff
@@ -113,4 +118,107 @@ public void sendMessage_shouldSendMessage() throws MessageException {
...
+		
+		assertNotNull(ms.getMessagePreparator());
+		
+	}
+	
+	/**
+	 * @see MessageService#setMessagePreparator(MessagePreparator)
+	 */
+	@Test
+	public void setMessagePreparator_shouldSetMessagePreparator(){
```
Comment (teleivo) [inline api/src/test/java/org/openmrs/notification/MessageServiceTest.java]: remove test

File: api/src/test/java/org/openmrs/notification/MessageServiceTest.java [hunk 5]
```diff
@@ -113,4 +118,107 @@ public void sendMessage_shouldSendMessage() throws MessageException {
...
+		assertNull(ms.getMessagePreparator());
+		
+	}
+	
+	/**
+	 * @throws MessageException
+	 * @see MessageService#prepareMessage(Template)
+	 */
+	@Test
+	public void prepareMessage_shouldPrepareMessage() throws MessageException{
```
Comment (teleivo) [inline api/src/test/java/org/openmrs/notification/MessageServiceTest.java]: lets work together on the other PR before we get into this one

**Comment by teleivo** [comment]
I strongly suggest that we first finish our work on #2536 as there is still a lot to do. This way we can work together so you get a better understanding of what and how we test. I feel that it does not make sense to take on more testing issues looking at this PR. Please dont feel discouraged, I'll work with you on the other PR. I am simply convinced from my past experience that focusing on one thing is best.

File: api/src/test/java/org/openmrs/notification/MessageServiceTest.java (hunk 5 above)
Comment (dilekamadushan) [reply api/src/test/java/org/openmrs/notification/MessageServiceTest.java]: @teleivo The coveralls showed up in red for these methods, and that was why I wrote  test methods. Thank you very much for your reviews.

**Pull request closed without merging**